src/sources/Source.js

```diff
@@ -208,6 +208,28 @@ class Source {
   getLayout() {
     return this.eeprom.LAYOUT;
   }
+
+  /**
+   * Get if the layout selection shall be disabled
+   * 
+   * @param {object} flash 
+   * 
+   * @returns {boolean}
+   */
+  getDisabledLayoutSelection(flash) {
+    return false;
+  }
+
+  /**
+   * Get the ESC name
+   * 
+   * @param {object} flash
+   * 
+   * @returns {string|null}
+   */
+  getEscName(flash) {
```

Comment:
Can you not override the `getLayout` method here? Or how is Layout handled differently than the Name?

Reply:
The source is a base, and has nothing to do with the ESC itself
I changed the code anyway and display the firmware name from eeprom here.